protocols/gossipsub/src/protocol.rs

```diff
@@ -397,3 +548,51 @@ pub enum GossipsubControlAction {
         topic_hash: TopicHash,
     },
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn sign_and_verify_message_peer_inline() {
```

Comment:
I think this could greatly benefit of random test input generation e.g. via quickcheck. I am happy to provide a commit in case you agree.

Reply:
Yeah I agree, I'm about to go off and build out 1.1. If you have the time to add this in, it would be great

Reply:
@AgeManning see https://github.com/sigp/rust-libp2p/pull/31.